syntax = "proto2";

package airspan_challenge;

message UserDetails {
  required string first_name = 1;
  required string last_name = 2;
  required string email = 3;
  optional string phone_number = 4;
}

message User {
  required string id = 1;
  required UserDetails UserDetails = 2;

}


message Users {
  repeated User people = 1;
}

message UserDelete {
  required string id = 1;
}


message UserUpdate {
  optional string first_name = 1;
  optional string last_name = 2;
  optional string email = 3;
  required string id = 4;
  optional string phone_number = 5;
}

message EditUserResponse {

    enum EditUserResult {
    FAIL = 0;
    SUCCESS = 1;
    NOTEXIST = 2;
  }

  required EditUserResult result = 1 [default = SUCCESS];
}

message WrapperMessage {
    required int64 commonField = 1;
    oneof msg {
        UserUpdate m1 = 2;
        User m2 = 3;
	UserDelete m3 = 4;
    }   
}
